### DISCUSSION TITLE
Detect model changes in onModelAfterUpdate hook

### DISCUSSION THREAD

**Discussion by lbensaad** [pocketbase/pocketbase#5792] [Q&A] [answered]
I need to create a payment when order status changed to 6 (delivered)
In sqlite after update trigger you have the NEW and OLD records, but in pb I could access both NEW and OLD model in onModelAfterUpdate, so i used onModelBeforeUpdate to store the old status in the model.
Is this the correct way:
```

onModelBeforeUpdate((e) => {
  const id = e.model.get('id');
  const oldModel = e.dao.findRecordById('orders', id);
  const old_status = oldModel.get('status');
  // put the old status in the model
  e.model.set('old_status', old_status);
}, 'orders');

onModelAfterUpdate((e) => {
  // get the old status
  const old_status = e.model.get('old_status');
  // get the new status
  const new_status = e.model.getInt('status');
  //console.log('order status', new_status);

  // order status is changed to delivered
  if (new_status === 6 && old_status !== 6) { 
    // create payment for order
  }
}, 'orders');
```

Or there is a better standard way? (i want to create the payment after the status update and not before)

**Comment by ganigeorgiev** [comment, marked as answer]
That's one possible approach, but another could be to call [`const old_status = e.model.originalCopy().get("status)`](https://pocketbase.io/jsvm/interfaces/models.Record.html#originalCopy).

**Reply by lbensaad** [reply]
Thanks, that is what I needed. Now I do not need `onModelBeforeUpdate` at all.

**Reply by lbensaad** [reply]
in the link you provided i see that `originalCopy()` is defined on Record but you called on `e.model` ?!

**Reply by ganigeorgiev** [reply]
`e.model` could be any Go struct that satisfies the `Model` interface (for collection record changes that is the `Record` struct, aka. `e.model` is `Record`). With v0.23.0-rc there are dedicated `OnRecord` db hooks to minimize the confusion and to avoid the extra type assertions that are required when extending with Go.

**Reply by lbensaad** [reply]
i started using v0.23.0-rc in dev, is there online docs for it?

**Reply by ganigeorgiev** [reply]
No sorry, I'm still working on it.
